docs/adr/0001-tenants-will-not-inherit-their-parents-properties.md

```diff
@@ -0,0 +1,39 @@
+# Tenants will not inherit their parents properties
+
+## Status
+
+Proposed
+
+## Context
+
+Corvus.Tenancy supports a hierarchy of tenants. There are two things that this can be used for:
+
+Firstly, we can control on a per-tenant basis where the children of that tenant are stored. For example, with two sibling tenants A and B (i.e. tenants that are children of the same parent tenant), the children of A can be stored in a completely different location to the children of B. By default, this will be a separate container in the same storage account, but it could be a completely separate storage account.
+
+Secondly, it can be used to enable better organisation of tenants by using parent tenants to group related tenants together.
+
+One of of the functions of tenants is to hold client-specific configuration for the applications that a client is using. An example would be for a client using the Workflow service. Their tenant will contain two pieces of storage information, one for Workflows and one for Workflow instances. This configuration is stored in a collection of key-value pairs attached to the tenant.
+
+It is possible for tenants to have child tenants in the hierarchy. If a tenant that uses the Workflow service has children, they may also need to use the Workflow service. In this case we have a choice: we can decide that we will allow the workflow storage configuration from a tenant to be inherited by its children, or we can require each tenant to contain all of it's own configuration.
+
+## Decision
+
+We have determined that we will not allow properties of tenants to be inherited by their children.
```

Comment:
"allow" seems slightly like the wrong word here. `Corvus.Tenancy` won't prevent me from writing something that deliberately copies all the properties down when creating new tenants, or when modifying existing ones, so it's not really disallowing it. It's just providing me with absolutely no support.

So I think I'd rephrase this along the lines that `Corvus.Tenancy` does not provide any means by which properties of tenants are inherited by their children. (It doesn't do it. But equally it wouldn't stop you from doing it if you don't mind doing all the relevant work. E.g., an implementation of `ITenantProvider` could, if it wanted, automatically copy the storage settings being used for the tenancy storage into child tenants by default, and `Corvus.Tenancy` allows this in the sense that it won't stop it.)

Reply:
(Indeed, you go on to say that later. So "not allow" isn't quite right.)